From 5f3bb5602615894cda88ca1b44fdfafdfb01c8c8 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Sun, 30 Jul 2023 11:20:39 +0200
Subject: [PATCH 3/5] treewide: use correct type for tasklet_init

Update and use correct type for tasklet_init to fix compilation error
for not valid cast.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 hal/hal_hci/hal_usb.c             | 2 +-
 hal/rtl8812a/usb/rtl8812au_xmit.c | 2 +-
 hal/rtl8812a/usb/usb_ops_linux.c  | 2 +-
 include/rtl8812a_xmit.h           | 2 +-
 include/usb_ops_linux.h           | 2 +-
 os_dep/linux/usb_ops_linux.c      | 4 ++--
 6 files changed, 7 insertions(+), 7 deletions(-)

--- a/hal/hal_hci/hal_usb.c
+++ b/hal/hal_hci/hal_usb.c
@@ -35,7 +35,7 @@ int	usb_init_recv_priv(_adapter *padapte
 
 #ifdef PLATFORM_LINUX
 	tasklet_init(&precvpriv->recv_tasklet,
-	             (void(*)(unsigned long))usb_recv_tasklet,
+	             usb_recv_tasklet,
 	             (unsigned long)padapter);
 #endif /* PLATFORM_LINUX */
 
--- a/hal/rtl8812a/usb/rtl8812au_xmit.c
+++ b/hal/rtl8812a/usb/rtl8812au_xmit.c
@@ -30,7 +30,7 @@ s32	rtl8812au_init_xmit_priv(_adapter *p
 
 #ifdef PLATFORM_LINUX
 	tasklet_init(&pxmitpriv->xmit_tasklet,
-	             (void(*)(unsigned long))rtl8812au_xmit_tasklet,
+	             rtl8812au_xmit_tasklet,
 	             (unsigned long)padapter);
 #endif
 #ifdef CONFIG_TX_EARLY_MODE
--- a/hal/rtl8812a/usb/usb_ops_linux.c
+++ b/hal/rtl8812a/usb/usb_ops_linux.c
@@ -475,7 +475,7 @@ _exit_recvbuf2recvframe:
 }
 
 
-void rtl8812au_xmit_tasklet(void *priv)
+void rtl8812au_xmit_tasklet(unsigned long priv)
 {
 	int ret = _FALSE;
 	_adapter *padapter = (_adapter*)priv;
--- a/include/rtl8812a_xmit.h
+++ b/include/rtl8812a_xmit.h
@@ -331,7 +331,7 @@ s32 rtl8812au_hal_xmit(PADAPTER padapter
 s32 rtl8812au_mgnt_xmit(PADAPTER padapter, struct xmit_frame *pmgntframe);
 s32	 rtl8812au_hal_xmitframe_enqueue(_adapter *padapter, struct xmit_frame *pxmitframe);
 s32 rtl8812au_xmit_buf_handler(PADAPTER padapter);
-void rtl8812au_xmit_tasklet(void *priv);
+void rtl8812au_xmit_tasklet(unsigned long priv);
 s32 rtl8812au_xmitframe_complete(_adapter *padapter, struct xmit_priv *pxmitpriv, struct xmit_buf *pxmitbuf);
 #endif
 
--- a/include/usb_ops_linux.h
+++ b/include/usb_ops_linux.h
@@ -78,7 +78,7 @@ int usb_write16(struct intf_hdl *pintfhd
 int usb_write32(struct intf_hdl *pintfhdl, u32 addr, u32 val);
 int usb_writeN(struct intf_hdl *pintfhdl, u32 addr, u32 length, u8 *pdata);
 u32 usb_read_port(struct intf_hdl *pintfhdl, u32 addr, u32 cnt, u8 *rmem);
-void usb_recv_tasklet(void *priv);
+void usb_recv_tasklet(unsigned long priv);
 
 #endif
 
--- a/os_dep/linux/usb_ops_linux.c
+++ b/os_dep/linux/usb_ops_linux.c
@@ -717,7 +717,7 @@ void usb_init_recvbuf(_adapter *padapter
 int recvbuf2recvframe(PADAPTER padapter, void *ptr);
 
 #ifdef CONFIG_USE_USB_BUFFER_ALLOC_RX
-void usb_recv_tasklet(void *priv)
+void usb_recv_tasklet(unsigned long priv)
 {
 	struct recv_buf *precvbuf = NULL;
 	_adapter	*padapter = (_adapter*)priv;
@@ -870,7 +870,7 @@ u32 usb_read_port(struct intf_hdl *pintf
 }
 #else	// CONFIG_USE_USB_BUFFER_ALLOC_RX
 
-void usb_recv_tasklet(void *priv)
+void usb_recv_tasklet(unsigned long priv)
 {
 	_pkt			*pskb;
 	_adapter		*padapter = (_adapter*)priv;
